src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicWorkspacePullDiagnosticsHandler.cs

```diff
@@ -43,10 +43,10 @@ internal class PublicWorkspacePullDiagnosticsHandler : AbstractPullDiagnosticHan
         IGlobalOptionService globalOptions)
         : base(analyzerService, editAndContinueDiagnosticUpdateSource, globalOptions)
     {
+        _workspaceManager = workspaceManager;
         _workspaceRegistrationService = registrationService;
-        _workspaceRegistrationService.LspSolutionChanged += OnLspSolutionChanged;
 
-        _workspaceManager = workspaceManager;
+        _workspaceRegistrationService.LspSolutionChanged += OnLspSolutionChanged;
```

Comment:
changed to hookup events *last* after all state has been assigned in constructor.  My brain can't handle the idea of possibly getting an event before we're fully initialized.